### PR TITLE
Gcds search

### PR DESCRIPTION
**Pull request by ethanWallace**
# Summary | Résumé

Add `gcds-search` to component library. `gcds-search` will render with all default values to meet Canada.ca standards. Form and input values can be customized to suit individual site needs.

``` html
<gcds-search
  placeholder="Canada.ca"        
  action="/sr/srb.html"    
  method="get"      
  name="q"    
  search-id="search"  
  suggested="" 
>
</gcds-search>
```

### Placeholder
Set the placeholder and hidden label for the search input. Will default to “Canada.ca” to match policy for global search. This will render as “Search [placeholder]” / ”Rechercher dans [placeholder]”.
### Action
Set the search form action. Will default to “sr/srb.html” to match policy for global search. The current language will be included to the full url to link to the appropriate language.
### Method
Set the method the search form uses. Will default to “get” to match policy for global search.
### Name
Set the name of the search input. Will default to “q” to match the global Canada.ca search.
### Search-id
Set the id of the search input. Will default to “search” if not passed.
### Suggested
Set the options for suggested search terms. This will populate a datalist attached to the search input

Uses tokens from https://github.com/cds-snc/gcds-tokens/pull/138

### CODE REVIEW

File: packages/web/src/components/gcds-header/gcds-header.css [hunk 1]
```diff
@@ -59,7 +58,8 @@
 
         .brand__search,
         slot[name="search"] {
-          width: var(--gcds-header-brand-search-width);
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-header/gcds-header.css:62]: Can we remove this token from the header tokens?

File: packages/web/src/components/gcds-search/gcds-search.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  /**
+  * Set the name of the search input
+  */
+  @Prop() searchId: string = "search";
+
+  /**
+  * Set a list of predefined search terms
+  */
+  @Prop() suggested: Array<string>;
+
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-search/gcds-search.tsx:43]: Can you add a comment here marking the start of the events? Like here:
https://github.com/cds-snc/gcds-components/blob/main/packages/web/src/components/gcds-button/gcds-button.tsx#L143

File: packages/web/src/components/gcds-search/stories/overview.mdx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+<Canvas
+  of={Search.Default}
+  story={{ inline: true }}
+/>
+
+## Examples
+<br/>
+
+### Default
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-search/stories/overview.mdx:20]: Can we add a few more examples?

For example, what does the default look like in French? Stepper has language examples:
https://cds-snc.github.io/gcds-components/?path=/docs/components-stepper--overview

File: packages/web/src/components/gcds-search/gcds-search.css [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+@layer defaults {
+  :host {
+    .gcds-search {
+      .gcds-search__header {
+        clip: rect(1px,1px,1px,1px);
+        height: 1px;
+        margin: 0;
+        overflow: hidden;
+        position: absolute;
+        width: 1px;
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-search/gcds-search.css:12]: Maybe a bit nit-picky but we have code in the [label](https://github.com/cds-snc/gcds-components/blob/main/packages/web/src/components/gcds-label/gcds-label.css#L9) component + the UF to hide text - can we use the same code here to be consistent?

```
  display: block;
  width: 0;
  height: 0;
  margin: 0;
  overflow: hidden;
```

File: packages/web/src/components/gcds-search/gcds-search.css [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        height: auto;
+        min-height: var(--gcds-search-min-width-and-height);
+        font: var(--gcds-search-font);
+        padding: var(--gcds-search-padding) !important;
+        background-color: var(--gcds-search-default-background);
+        background-image: none;
+        color: var(--gcds-search-default-text);
+        border: var(--gcds-search-border-width) solid currentColor;
+        border-radius: var(--gcds-search-border-radius);
+        border-top-right-radius: 0;
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-search/gcds-search.css:31]: Can you change `border-top` to `border-block-start` to be more language inclusive?

File: packages/web/src/components/gcds-search/gcds-search.css [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        border-bottom-right-radius: 0;
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-search/gcds-search.css:32]: Can you change `border-bottom` to `border-block-end` to be more language inclusive?

Comment (melaniebmn) [inline packages/web/src/components/gcds-search/gcds-search.css:32]: Can `border-top...` and `border-bottom...` be replaced with `border-radius: 0;`? Looks like the whole input field is not supposed to have a border radius.

Comment (melaniebmn) [inline packages/web/src/components/gcds-search/gcds-search.css:32]: Can you replace `border-bottom-right-radius` with `border-end-end-radius` to be more language inclusive?

File: packages/web/src/components/gcds-search/gcds-search.css (hunk 5 above)
Comment (melaniebmn) [inline packages/web/src/components/gcds-search/gcds-search.css:31]: Can you replace `border-top-right-radius` with `border-start-end-radius` to be more language inclusive?

File: packages/web/src/components/gcds-search/gcds-search.css [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        box-sizing: border-box;
+        transition: border-color ease-in-out .15s, box-shadow ease-in-out .15s;
+      }
+
+      ::part(button) {
+        border-top-left-radius: 0;
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-search/gcds-search.css:38]: Can you replace `border-top-left-radius` with `border-start-start-radius` to be more language inclusive?

File: packages/web/src/components/gcds-search/gcds-search.css [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        border-bottom-left-radius: 0;
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-search/gcds-search.css:39]: Can you replace `border-bottom-left-radius` with `border-end-start-radius` to be more language inclusive?

File: packages/web/src/components/gcds-search/gcds-search.tsx [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    const attrsInput = {
+      name,
+      placeholder: labelText
+    };
+
+    let formAction = action === '/sr/srb.html' ? `https://www.canada.ca/${lang}/sr/srb.html` : action;
+
+    return (
+      <Host>
+        <section class="gcds-search">
```
Comment (melaniebmn) [inline packages/web/src/components/gcds-search/gcds-search.tsx:100]: How do you feel about replacing the `section` tag with a `div` tag? We didn't flag other parts like the breadcrumbs for example as sections in the header.

**Review by melaniebmn — CHANGES_REQUESTED** [review]
A few small comments/changes but overall looks good!

File: packages/web/src/components/gcds-search/gcds-search.tsx (hunk 9 above)
Comment (ethanWallace) [reply packages/web/src/components/gcds-search/gcds-search.tsx:100]: If we haven't used it in the other parts of the header, I will switch it to a `div`

**Review by melaniebmn — APPROVED** [review]
LGTM

**Pull request merged**